web/js/panel.js

```diff
@@ -678,13 +793,103 @@ function processPanelXML($returnedData, $success, $xhr) {
+
+//     if (null == $widget) {
+//         logProperties(this);
+//     }
+
+    // special case for layoutSlips
+    if (this.id.endsWith("r") || this.id.endsWith("l")) {
+        var slipID = this.id.slice(0, -1);
+        $widget = $gWidgets[slipID];
+
+        if (this.id.endsWith("r")) {
+            $widget["side"] = "left";
+        } else if (this.id.endsWith("l")) {
+            $widget["side"] = "right";
+        }
+        //jmri.log("  layoutSlip-side:" + $widget.side);
+
+        var $newState = $widget.state;
+
+        //jmri.log("  layoutSlip-slipType:" + $widget.slipType);
+        if ($widget.side == 'left') {
+            switch ($widget.state) {
+                case STATE_AC:
+                    if ($widget.slipType == DOUBLE_SLIP) {
+                        $newState = STATE_BC;
+                    } else {
+                        $newState = STATE_BD;
+                    }
+                    break;
+                case STATE_BD:
+                default:
+                    $newState = STATE_AD;
+                    break;
+                case STATE_AD:
+                    $newState = STATE_BD;
+                    break;
+                case STATE_BC:
+                    $newState = STATE_AC;
+                    break;
+            }
+        } else if ($widget.side == 'right') {
+            switch ($widget.state) {
+                case STATE_AC:
+                default:
+                    $newState = STATE_AD;
+                    break;
+                case STATE_BD:
+                    if ($widget.slipType == DOUBLE_SLIP) {
+                        $newState = STATE_BC;
+                    } else {
+                        $newState = STATE_AC;
+                    }
+                    break;
+                case STATE_AD:
+                    $newState = STATE_AC;
+                    break;
+                case STATE_BC:
+                    $newState = STATE_BD;
+                    break;
+            }
+        }
+
+        //jmri.log("$handleClick:layoutSlip: change state from: " + $widget.state + ", to: " + $newState + ".");
+        $widget.state = $newState;
+        $gWidgets[$widget.id].state = $newState;
+
+        var $newState1 = CLOSED;
+        var $newState2 = CLOSED;
+        switch ($newState) {
+            case STATE_AC:
+                $newState2 = THROWN;
+                break;
+            case STATE_BC:
+                $newState1 = THROWN;
+                $newState2 = THROWN;
+                break;
+            case STATE_AD:
+                break;
+            case STATE_BD:
+            default:
+                $newState1 = THROWN;
+                break;
+        }
+        sendElementChange($widget.jsonType, $widget.name, $newState1);
+        sendElementChange($widget.jsonType, $widget.name2nd, $newState2);
+        $drawSlip($widget); //draw the slip
```

Comment:
Ok, I'm probably missing something here… without that line I'm not getting the screen updates… probably same reason I'm not getting updates when I click slip circles in PanelPro's Layout Editor window.